### PR TITLE
Fix ISE while deleting a DS created from a DSR

### LINKED ISSUE #6549: Deleting a Delivery Service created as a result of a Delivery Service Request results in Internal Server Error
## This Bug Report affects these Traffic Control components:

- Traffic Ops

## Current behavior:

Run TO and TP.
Create a DSR and fulfill it.
Try deleting the resulting DS.
You will notice an ISE 500.

## Expected behavior:

Deleting a DS created from a DSR should not give an ISE 500.

## Steps to reproduce:

Create a DSR and fulfill it.
Then, try to delete the created DS.
You will see an internal server error with the following log lines in error.log:

```
ERROR: server.go:3170: http: superfluous response.WriteHeader call from github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/routing/middleware.GzipResponse (wrappers.go:240)
ERROR: api.go:253: 2022-02-09T16:32:31.107982633Z: 10.234.97.13:40522 panic: (err: runtime error: invalid memory address or nil pointer dereference) stacktrace:
goroutine 5499473 [running]:
github.com/apache/trafficcontrol/lib/go-util.Stacktrace()
        /tmp/go/src/github.com/apache/trafficcontrol/lib/go-util/util.go:32 +0x66
github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/routing/middleware.WrapPanicRecover.func1.1()
        /tmp/go/src/github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/routing/middleware/wrappers.go:173 +0x66
panic({0xe000c0, 0x16ab940})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/deliveryservice/request.PutAssignment({0x1079830, 0xc004c227b0}, 0xc001320c00)
        /tmp/go/src/github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/deliveryservice/request/assign.go:226 +0xa8d
github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/routing/middleware.RequiredPermissionsMiddleware.func1.1({0x1079830, 0xc004c227b0}, 0xc001320c00)
        /tmp/go/src/github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/routing/middleware/wrappers.go:312 +0x4cf
github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/routing/middleware.AuthBase.GetWrapper.func1.1({0x1079830, 0xc004c227b0}, 0xc001320c00)
        /tmp/go/src/github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/routing/middleware/wrappers.go:127 +0x32d
github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/routing/middleware.WrapPanicRecover.func1({0x1079830, 0xc004c227b0}, 0xf004a6)
        /tmp/go/src/github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/routing/middleware/wrappers.go:177 +0x7e
github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/routing/middleware.WrapHeaders.func1({0x107bdb0, 0xc0012aa9c0}, 0x6b769f)
        /tmp/go/src/github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/routing/middleware/wrappers.go:157 +0x60d
net/http.HandlerFunc.ServeHTTP(0x1, {0x107bdb0, 0xc0012aa9c0}, 0x43ef45)
        /usr/local/go/src/net/http/server.go:2047 +0x2f
net/http.(*timeoutHandler).ServeHTTP.func1()
        /usr/local/go/src/net/http/server.go:3337 +0x7c
created by net/http.(*timeoutHandler).ServeHTTP
        /usr/local/go/src/net/http/server.go:3331 +0x465

goroutine 1 [chan receive, 11790 minutes]:
main.signalReloader({0x1079630, 0x1060770}, 0xc000375be0)
        /tmp/go/src/github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/traffic_ops_golang.go:379 +0xc7
main.main()
        /tmp/go/src/github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/traffic_ops_golang.go:236 +0x1253

goroutine 3 [select, 11790 minutes]:
database/sql.(*DB).connectionOpener(0xc00032e680, {0x1083738, 0xc00030c000})
        /usr/local/go/src/database/sql/sql.go:1196 +0x93
```

### PR DESCRIPTION
**Pull request by srijeet0406**
This PR closes #6549 

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

Follow the steps in #6549 
## If this is a bugfix, which Traffic Control versions contained the bug?

- master

## PR submission checklist
- [ ] This PR has tests 
- [ ] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/deliveryservice/request/assign.go [hunk 1]
```diff
@@ -223,9 +223,11 @@ func PutAssignment(w http.ResponseWriter, r *http.Request) {
 	}
 	var resp interface{}
 	if inf.Version.Major >= 4 {
-		if dsr.Requested.LongDesc1 != nil || dsr.Requested.LongDesc2 != nil {
-			api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New("the longDesc1 and longDesc2 fields are no longer supported in API 4.0 onwards"), nil)
-			return
+		if dsr.Requested != nil {
+			if dsr.Requested.LongDesc1 != nil || dsr.Requested.LongDesc2 != nil {
```
Comment (zrhoffman) [inline traffic_ops/traffic_ops_golang/deliveryservice/request/assign.go:227]: Nit: these 2 _if_ statements can be combined

**Review by zrhoffman — CHANGES_REQUESTED** [review]

File: traffic_ops/traffic_ops_golang/deliveryservice/request/assign.go [hunk 2]
```diff
@@ -223,12 +223,11 @@ func PutAssignment(w http.ResponseWriter, r *http.Request) {
 	}
 	var resp interface{}
 	if inf.Version.Major >= 4 {
-		if dsr.Requested != nil {
-			if dsr.Requested.LongDesc1 != nil || dsr.Requested.LongDesc2 != nil {
-				api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New("the longDesc1 and longDesc2 fields are no longer supported in API 4.0 onwards"), nil)
-				return
-			}
+		if dsr.Requested != nil && dsr.Requested.LongDesc1 != nil || dsr.Requested.LongDesc2 != nil {
```
Comment (zrhoffman) [inline traffic_ops/traffic_ops_golang/deliveryservice/request/assign.go:226]:
```go
dsr.Requested != nil && dsr.Requested.LongDesc1 != nil || dsr.Requested.LongDesc2 != nil
```
is not equal to
```go
dsr.Requested != nil && (dsr.Requested.LongDesc1 != nil || dsr.Requested.LongDesc2 != nil)
```
for
* `dsr.Requested != nil` => false, `dsr.Requested.LongDesc1 != nil` => false, `dsr.Requested.LongDesc2 != nil` => true
* `dsr.Requested != nil` => false, `dsr.Requested.LongDesc1 != nil` => true, `dsr.Requested.LongDesc2 != nil` => true

**Review by zrhoffman — CHANGES_REQUESTED** [review]

File: traffic_ops/traffic_ops_golang/deliveryservice/request/assign.go (hunk 2 above)
Comment (srijeet0406) [reply traffic_ops/traffic_ops_golang/deliveryservice/request/assign.go]: I was reading the if statement wrong, fixed now.

**Review by zrhoffman — APPROVED** [review]
Looks good!

**Pull request merged**